Userland/Libraries/LibGfx/PNGLoader.cpp

```diff
@@ -47,8 +47,8 @@ struct CodingIndependentCodePoints {
 static_assert(AssertSize<CodingIndependentCodePoints, 4>());
 
 struct EmbeddedICCProfile {
-    StringView profile_name;
-    ReadonlyBytes compressed_data;
+    ByteBuffer profile_name;
+    ByteBuffer compressed_data;
```

Comment:
I don't think this change is needed? This is ephemeral state during decoding. The buffer that's handed out to clients is the decompressed one.

Reply:
that's needed because reading png chunks with `Stream::read()` requires us to make a buffer. the icc decompression and parsing takes place after parsing chunks, thus we need to store the data somewhere for that time being.

if we don't really want to store the buffer then maybe we could decompress and parse icc directly when reading iCCP blocks?

another possible solution would be to use FixedMemoryStream's `stream.bytes().slice(stream.offset(), block_length); TRY(stream.discard(block_length));` in `process_chunk()` to avoid memory allocation (and it is what i did in scanline decoding in `decode_png_bitmap_simple()`) but i feel like it kind of misses the point of the whole 'streaming' thing..

<sup><sub>that being said, i don't want to work on this pr anymore.. i wrote this message because i felt bad for not answering you</sub></sup>